plugins/spu2-x/src/Mixer.cpp

```diff
@@ -294,42 +279,15 @@ static s32 __forceinline GetNoiseValues()
 		retval = (Seed&0xff) << 8;
 	else if( Seed&0xffff )
 		retval = 0x7fff;
-#ifdef _WIN32
-	__asm {
-		MOV eax,Seed
-		ROR eax,5
-		XOR eax,0x9a
-		MOV ebx,eax
-		ROL eax,2
-		ADD eax,ebx
-		XOR eax,ebx
-		ROR eax,3
-		MOV Seed,eax
-	}
-#elif !defined(__clang__) // Linux with GCC
-	__asm__ (
-		".intel_syntax\n"
-		"MOV %%eax,%1\n"
-		"ROR %%eax,5\n"
-		"XOR %%eax,0x9a\n"
-		"MOV %%esi,%%eax\n"
-		"ROL %%eax,2\n"
-		"ADD %%eax,%%esi\n"
-		"XOR %%eax,%%esi\n"
-		"ROR %%eax,3\n"
-		"MOV %0,%%eax\n"
-		".att_syntax\n" : "=r"(Seed) :"r"(Seed)
-		: "%eax", "%esi"
-		);
-#else // Clang and others
-	s32 s = rotr(Seed,5);
-	s ^= 0x9a;
-	s32 k = rotl(s,2);
-	k+=s;
-	k^=s;
-	k = rotr(k,3);
-	Seed=k;
```

Comment:
> Maybe it would have been better just to modify this version slightly?

I don't understand, how would it be better ? I only see the disadvantage on maintaining unnecessary platform specific codes.